### PR TITLE
HistogramData rollout: algorithms EnggDiffFittingPresenter.cpp to MuonAnalysisHelperTest.h

### PR DESCRIPTION
**Pull request by LamarMoore**
## Description of work.
This PR is part of refactoring effort to use the new interface of the HistogramData module.
See https://github.com/mantidproject/mantid/issues/17641 for details.

The following algorithms were refactored to use HistogramData:
MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp
MantidQt/CustomInterfaces/src/MantidEVWorker.cpp
MantidQt/CustomInterfaces/src/MultiDatasetFit/MDFDataController.cpp
MantidQt/CustomInterfaces/src/MultiDatasetFit/MDFDatasetPlotData.cpp
MantidQt/CustomInterfaces/src/SampleTransmission.cpp
MantidQt/CustomInterfaces/src/SANSPlotSpecial.cpp
MantidQt/CustomInterfaces/src/SANSRunWindow.cpp
MantidQt/CustomInterfaces/src/Tomography/ImageROIViewQtWidget.cpp
MantidQt/CustomInterfaces/src/Tomography/ImggFormatsConvertViewQtWidget.cpp
MantidQt/CustomInterfaces/src/Tomography/TomographyIfaceViewQtGUI.cpp
MantidQt/CustomInterfaces/test/ALCBaselineModellingModelTest.h
MantidQt/CustomInterfaces/test/ALCBaselineModellingPresenterTest.h
MantidQt/CustomInterfaces/test/ALCPeakFittingModelTest.h
MantidQt/CustomInterfaces/test/MuonAnalysisDataLoaderTest.h
MantidQt/CustomInterfaces/test/MuonAnalysisHelperTest.h

## To test.

Code review.

<Add any manual testing not covered by unit tests, especially GUIs>

Fixes <your issue>.
Internal change, no release notes (potential performance improvements will be added to release notes as part of the umbrella issue).

### CODE REVIEW

File: MantidQt/CustomInterfaces/src/SampleTransmission.cpp [hunk 1]
```diff
@@ -184,8 +184,8 @@ void SampleTransmission::algorithmComplete(bool error) {
   m_uiForm.twResults->addTopLevelItem(transmissionItem);
   transmissionItem->setExpanded(true);
 
-  std::vector<double> transmissionData = ws->dataY(0);
-  Statistics stats = getStatistics(transmissionData);
+  auto &transmissionData = ws->y(0);
+  Statistics stats = getStatistics(transmissionData.rawData());
```
Comment (DimitarTasev) [inline MantidQt/CustomInterfaces/src/SampleTransmission.cpp]: `getStatistics(ws->y(0).rawData())`, the transmissionData variable is not used below

**Review by dtasev — COMMENTED** [review]
Just a minor comment, everything else looks good

**Comment by DimitarTasev** [comment]
Changes look good, if @SimonHeybrock doesn't have anything to add, I'm happy to :shipit:

**Comment by martyngigg** [comment]
Have all of the GUI interfaces here been manually tested?

**Comment by martyngigg** [comment]
Feel free to enlist the experts on those interfaces to help test drive them.

**Comment by DimitarTasev** [comment]
I'll have a look at Engineering Diffraction and Tomography interfaces

**Comment by DimitarTasev** [comment]
I haven't found any regression in Engineering Diff and Tomography interfaces functionality

**Comment by LamarMoore** [comment]
I tested the Muon Analysis interfaces. Seemed to work fine.

**Comment by martyngigg** [comment]
@AntonPiccardoSelg Could check over the changes to SANS?

**Comment by AntonPiccardoSelg** [comment]
There is no obvious issue when  interacting with the GUI and touching the areas which were changed by this PR. The SANS changes seem to be fine.

**Comment by martyngigg** [comment]
I've checked the multidataset fitting interface and that looks okay too.

Thanks to everyone for checking all that out.

**Pull request merged**